testcase/shellstatus.mk

```diff
@@ -0,0 +1,28 @@
+# Check the value of .SHELLSTATUS before $(shell) has run
+A := $(.SHELLSTATUS)
+
+# True always returns 0
+$(shell true)
+B := $(.SHELLSTATUS)
+
+# False always returns 1
+$(shell false)
+C := $(.SHELLSTATUS)
+
+# .SHELLSTATUS is global across makefiles
+$(file >nested.mk,$$(shell true))
+include nested.mk
+D := $(.SHELLSTATUS)
+
+# Assigning to .SHELLSTATUS should have no effect
+.SHELLSTATUS := 3
+E := $(.SHELLSTATUS)
+
+# Test that you can't bypass the readonlyness of .SHELLSTATUS
+NAMEWORKAROUND := .SHELLSTATUS
+$(NAMEWORKAROUND) := 5
+F := $(.SHELLSTATUS)
+
+
+test:
+	echo $(A) $(B) $(C) $(D) $(E) $(F)
```

Comment:
Where is this code executed during testing and where is the expected value specified?

Reply:
The kati test harness runs the test against both kati and gnu make, and will fail if there are any differences in the output.